include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_cshuffle_streamk_v3.hpp

```
@@ -147,8 +148,15 @@ struct DeviceGemm_Xdl_CShuffle_Streamk_V3 : public DeviceGemm_Streamk_V2<ALayout
             index_t K_split = (arg.K + k_grain - 1) / k_grain * KPerBlock;
 
             const bool has_main_k_block_loop = GridwiseGemm::CalculateHasMainKBlockLoop(K_split);
-            hipGetErrorString(hipMemsetAsync(
-                arg.p_c_grid, 0, arg.M * arg.N * sizeof(CDataType), stream_config.stream_id_));
+
+            if constexpr(GridwiseGemm::Block2CTileMap_streamk::ReductionStrategy ==
+                         StreamKReductionStrategy::Atomic)
+            {
+
+                hipGetErrorString(hipMemsetAsync(
```

Does hipGetErrorString affect kernel performance here?

---

hip error checks happen on the host and hence does not affect kernel performance. Further, it is used in a few other places in the repo in host-side code. So, I think it should be okay. In my revision, I have modified all occurrences in the file for uniformity.
Additionally, for some of the hip error checks I have in this file, compiler gives me an error if I don't check return value and the only way around seems to be to suppress these errors using compiler flags which may not be preferable.